hydra-node/src/Hydra/Chain/ZeroMQ.hs

```diff
@@ -106,14 +105,14 @@ transactionPublisher chainSyncAddress txQueue tracer = do
     liftIO $ traceWith tracer (PublishingTransaction tx)
     send pub [] (Enc.encodeUtf8 $ show tx)
 
-transactionSyncer :: Tx tx => String -> TVar [OnChainTx tx] -> Tracer IO (MockChainLog tx) -> ZMQ z ()
+transactionSyncer :: Tx tx => String -> TVar IO [OnChainTx tx] -> Tracer IO (MockChainLog tx) -> ZMQ z ()
 transactionSyncer chainCatchupAddress transactionLog tracer = do
   rep <- socket Rep
   bind rep chainCatchupAddress
   liftIO $ traceWith tracer (TransactionSyncerStarted chainCatchupAddress)
   forever $ do
     _ <- receive rep
-    txs <- liftIO $ readTVarIO transactionLog
+    txs <- liftIO $ atomically $ readTVar transactionLog
```

Comment:
This will trigger HLint right? .. We could define such lifted helpers in the `Hydra.Prelude` now (and PR it at `io-sim-classes`?)

Reply:
It'll indeed and yes, I thought of adding it but :man_shrugging: